### PR TITLE
feat: Add a poetry-specific parser, and a mechanism for selecting it.

### PR DESCRIPTION
**Pull request by DanCardin**
Fixes https://github.com/sphinx-toolbox/sphinx-pyproject/issues/29

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -189,6 +189,7 @@ plugins = coverage_pyver_pragma
 
 [coverage:report]
 fail_under = 100
+show_missing = true
```
Comment (DanCardin) [inline tox.ini:192]: I was having trouble figuring out which lines were missing coverage, given the requirement that it be 100% coverage.

File: sphinx_pyproject/__init__.py [hunk 2]
```diff
@@ -273,10 +285,45 @@ def parse(
...
+		"""
+		Parse poetry's authors key.
+
+		:param config: The unparsed TOML config for the ``[tool.poetry]`` table.
+		"""
+
+		pep621_style_authors: List[dict[str, str]] = []
+
+		for author in config["author"]:
+			match = re.match(r"(?P<name>.*)<(?P<email>.*)>", author)
```
Comment (DanCardin) [inline sphinx_pyproject/__init__.py:317]: I originally implemented this as a straight-copy of the base class's `parse_author`. The implementation used a simpler string splitting mechanism to **just** pull out the name, which is the required field for "author", iirc.

But given the 100% coverage requirement, I wanted to get reuse out of the preexisting test that handles lack of any authors at all, so it felt perhaps better to redirect back to the base class's implementation ultimately.

But I could just as easily add a test and have the simpler impl here.

**Comment by keriksson-rosenqvist** [comment]
Is there anything happening with this?

**Comment by DanCardin** [comment]
For what it's worth, i've personally just been depending upon my fork's above revision in the interim

**Comment by domdfcoding** [comment]
Sorry for the wait. This went under the radar.

I've rebased and added some more documentation and one extra test. Once the tests are green I'll get this merged and released.

**Review by domdfcoding — APPROVED** [review]

**Comment by DanCardin** [comment]
thanks!

**Pull request merged**